torch/csrc/jit/script/init.cpp

```diff
@@ -25,6 +25,8 @@ struct VISIBILITY_HIDDEN PythonValue : public SugaredValue {
 
   // call it like a function, e.g. `outputs = this(inputs)`
   virtual std::shared_ptr<SugaredValue> call(SourceRange loc, Method & m, at::ArrayRef<Value*> inputs, List<Attribute> attributes, size_t n_binders) override {
+    ensureTensors(loc, inputs);
```

Comment:
Why do we do that? It doesn't really matter what do we call the PythonOp with, right?

Reply:
Tuples are not first-class in the interpreter (they must successfully lower in the tuple lowering pass). This means we can't pass or return them from python functions yet, otherwise the interpreter needs to know how to create/destructure them

Reply:
I understand that. My argument is more that it shouldn't be the responsibility of the compiler to check for those cases, but rather of the tuple lowering pass. Passing a tuple as an argument is something not amenable to lowering, and this is the place where we should error out. This will require fewer changes in the future.